### ISSUE TITLE
Updating to conform to JSON API v1?

### ISSUE THREAD

**Issue by thesteady** [RestPack/restpack_serializer#123]
Hi, @GavinJoyce! What are your thoughts on updating `restpack_serializer` to conform to new JSON API requirements? Do you plan to or want to update things? I'm happy to help out with the work if it needs to be done.

**Comment by GavinJoyce** [comment]
Yes, we should most certainly update to support JSON API 1.0 once it is stable. I'm hoping to have time again soon to contribute to RestPack::Serializer. Thanks for offering to help out, that is very much appreciated

**Comment by ms-ati** [comment]
:+1: If you are committed to fully implementing the standard (such as `?filter[id]=1,2&filter[name]=Gavin`), there are folks who will help

**Comment by steveklabnik** [comment]
We just did an RC3 release, to keep you updated.


https://github.com/json-api/json-api/issues/484

**Comment by thesteady** [comment]
Just as an update note, I've started a branch to start working on this: https://github.com/thesteady/restpack_serializer/tree/update_to_RC3_compliance

**Comment by fotinakis** [comment]
I'm starting to move away from another library that does JSON-API handling because I need more simplicity and flexibility, and was just about to build a pure serialization library—but this looks like exactly what I need!

@thesteady how far did you get in your fork for RC3 compliance and @GavinJoyce how much work do you think is involved in getting this mostly up to date? JSON-API 1.0 is landing this month and my timeframe is even shorter, so I probably will take a stab at this if no one else is working on it.

**Comment by thesteady** [comment]
Hi @fotinakis I've been focusing on other thing recently, and won't have time in the next two weeks to work on this. Feel free to go for it!

**Comment by fotinakis** [comment]
Whew, well I spent a little while trying to get this up to date with the current version of the spec, but honestly it didn't happen because the spec has changed so drastically in the last year (this library is really well-written though!).

I actually decided to just go all the way and write it myself over the last few days because I wanted a non-ActiveRecord version and needed some complex recursive includes and "linkage" handling for compound documents, so I did the crazy thing and wrote it from scratch. If you're interested, here it is!
https://github.com/fotinakis/jsonapi-serializers
